dds_web/utils.py

```diff
@@ -85,6 +85,23 @@ def verify_project_access(project) -> None:
         )
 
 
+def verify_project_user_key(project) -> None:
+    """Verify that current authenticated user has a row in projectUserKeys."""
+    project_key = models.ProjectUserKeys.query.filter_by(
+        project_id=project.id, user_id=auth.current_user().username
+    ).one_or_none()
+    if not project_key:
+        msg = (
+            "Access to all active projects has been lost, probably due to password reset. "
+            "Please, ask the corresponding unit to run dds project access fix"
+        )
+        raise AccessDeniedError(
+            message=msg,
+            username=auth.current_user().username,
+            project=project.public_id,
+        )
```

Comment:
This checks the row for a specific project and user, so we can't say that access to all active projects have been lost - we only know about this specific project. 

```suggestion
def verify_project_user_key(project) -> None:
    """Verify that current authenticated user has a row in projectUserKeys."""
    project_key = models.ProjectUserKeys.query.filter_by(
        project_id=project.id, user_id=auth.current_user().username
    ).one_or_none()
    if not project_key:
        msg = (
            "You have lost access to this project. This is likely due to a password reset, in which case you have lost access to all active projects. In order to regain access to this project, please contact {project.responsible_unit.external_display_name} and ask them to run 'dds project access fix'."
        )
        raise AccessDeniedError(
            message=msg,
            username=auth.current_user().username,
            project=project.public_id,
        )
```